compiler/src/iree/compiler/Dialect/Flow/TransformExtensions/FlowExtensions.cpp

```diff
@@ -576,12 +576,45 @@ transform_dialect::MovePrecedingOpIntoDispatchRegionOp::apply(
 //
 // All operands of the target are replaced with values defined inside of the
 // dispatch region when possible.
+static FailureOr<Operation *> cloneSucceedingOpIntoDispatchRegion(
+    RewriterBase &rewriter, Operation *target,
+    Flow::DispatchRegionOp regionOp) {
+  assert(regionOp->isBeforeInBlock(target) &&
```

Comment:
you want a nicer definitefailure if either not in the same block of not before in block rather than assert